--- callproc.c	2016-01-27 10:53:11.000000000 +0300
+++ callproc.c.2.7.mingw	2016-03-31 07:20:15.000000000 +0300
@@ -779,7 +779,7 @@
     int *space;
     ffi_cif cif;
     int cc;
-#ifdef MS_WIN32
+#if defined(MS_WIN32) && !defined(__MINGW32__)
     int delta;
 #ifndef DONT_USE_SEH
     DWORD dwExceptionCode = 0;
@@ -831,8 +831,10 @@
 #ifndef DONT_USE_SEH
     __try {
 #endif
+#ifndef __MINGW32__
         delta =
 #endif
+#endif
                 ffi_call(&cif, (void *)pProc, resmem, avalues);
 #ifdef MS_WIN32
 #ifndef DONT_USE_SEH
@@ -865,6 +867,7 @@
         return -1;
     }
 #endif
+#ifndef __MINGW32__
 #ifdef MS_WIN64
     if (delta != 0) {
         PyErr_Format(PyExc_RuntimeError,
@@ -895,6 +898,7 @@
     }
 #endif
 #endif
+#endif
     if ((flags & FUNCFLAG_PYTHONAPI) && PyErr_Occurred())
         return -1;
     return 0;
