--- dynload_win.c	2016-01-27 10:54:56.000000000 +0300
+++ dynload_win.c.3.5.mingw	2016-03-25 11:08:08.969689462 +0300
@@ -43,6 +43,7 @@
 /* Case insensitive string compare, to avoid any dependencies on particular
    C RTL implementations */
 
+#ifndef __MINGW32__
 static int strcasecmp (char *string1, char *string2)
 {
     int first, second;
@@ -56,7 +57,7 @@
 
     return (first - second);
 }
-
+#endif
 
 /* Function to return the name of the "python" DLL that the supplied module
    directly imports.  Looks through the list of imported modules and
